### ISSUE TITLE
proposal to change Travis file

### ISSUE THREAD

**Issue by mathematicalmichael** [UT-CHG/BET#322]
right now the current `.travis.yml` file [line 16](https://github.com/UT-CHG/BET/blob/master/.travis.yml#L16) follows the previous style of setting up, installing all the packages manually.

I propose that we remove any packages listed in this line that are explicitly stated in `setup.py`, which is called on the following line. I believe this would likely mirror a more common setup scenario, wherein a fresh env is created for isolation. So, just keep `mpi4py nose codecov` 

I dunno... not high priority. But thought I'd drop the idea here and get input on this proposal.

**Comment by smattis** [comment]
@mathematicalmichael I am on vacation and can think about this stuff next week. Also, I would very much like a chance to review your pull requests before they are merged. Please, in the future @ me in your pull requests, so I can review them. It might take a few days or a week, but I want to look at them. Please don't go around me.

**Comment by mathematicalmichael** [comment]
I didn’t, I just got a lot done over the weekend (I worked all day every day), and today was the first time I finally got troy down to sit and review code. Timing just aligned well. We went line by line. Helping make the diff smaller on big features coming down the line, which are actually worth your time to review. Entire methods and classes. Everything today was just small stuff to set the stage for an easier review. 

Sorry if you got that impression, it certainly wasn’t my intention. Troy and I have been discussing code changes for weeks and I finally got them done in a way that would have no impact on existing code. I’m hoping to submit the major PRs soon. One is at 95% coverage but is just missing docs. The other is new but more major, and I’m not sure what Troy told you about it. In any case, ensuring full backwards compatibility and adherence to the styling is a priority for me. As is efficiency with memory.
